tests/baselines/reference/dependentDestructuredVariables.errors.txt

```diff
@@ -1,8 +1,13 @@
 tests/cases/conformance/controlFlow/dependentDestructuredVariables.ts(334,5): error TS7022: 'value1' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.
 tests/cases/conformance/controlFlow/dependentDestructuredVariables.ts(334,5): error TS7031: Binding element 'value1' implicitly has an 'any' type.
+tests/cases/conformance/controlFlow/dependentDestructuredVariables.ts(408,7): error TS2322: Type '(type: "one" | "two", x: string | number | undefined) => void' is not assignable to type '(...args: [type: "one", x?: number | undefined] | [type: "two", x: string]) => void'.

```

Comment:
This is a case that regressed with [the newest commit](https://github.com/microsoft/TypeScript/pull/49218/commits/e3574503013a5dc983967a7ea30841ce4be3f7d9). I'm still thinking how to properly fix this - perhaps the solution is to disallow optional and required elements matching (as I mentioned in [this comment](https://github.com/microsoft/TypeScript/pull/49218/files#r1071104294)). It's a little bit fuzzy to me how optional elements followed but variable ones should behave, so I need to find a moment to write down some test cases and try out the idea with those in place.